### PR TITLE
Updates to Identity Resolver

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Fix up identity resolver to make it more reliable using `Build.RequestedForEmail` variable rather than `Build.QueuedBy`. https://github.com/Azure/azure-sdk-tools/issues/980
- Publish notification configuration tools to the azure-sdk-for-net DevOps feed.
- Combine CodeOwnersParser logic in a single library. https://github.com/Azure/azure-sdk-tools/issues/824

### CODE REVIEW

File: eng/common/pipelines/templates/steps/get-pr-owners.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ steps:
         --kusto-url-var KUSTO_URL `
         --kusto-database-var KUSTO_DB `
         --kusto-table-var KUSTO_TABLE `
-        --identity "$(Build.QueuedBy)" `
+        --identity "$(Build.RequestedForEmail)" `
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/get-pr-owners.yml:20]: We sure this works with github? Looks like it might be specific to Azure Repos, see https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#how-are-the-identity-variables-set.

Comment (chidozieononiwu) [reply eng/common/pipelines/templates/steps/get-pr-owners.yml:20]: May be I should use both username and email.

File: tools/code-owners-parser/CodeOwnersParser/CodeOwnersParser.csproj [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
```
Comment (weshaggard) [inline tools/code-owners-parser/CodeOwnersParser/CodeOwnersParser.csproj:4]: Not sure it matters but we might want to make this netstandard2.0 so we could consume this in other target frameworks if needed.

File: tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs [hunk 3]
```diff
@@ -40,12 +38,12 @@ public bool IsValid
 
         private static string[] SplitLine(string line, char splitOn)
         {
-            return line.Split(splitOn, StringSplitOptions.RemoveEmptyEntries);
+            return line.Split(new char[] { splitOn }, StringSplitOptions.RemoveEmptyEntries);
```
Comment (weshaggard) [inline tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs:41]: Why these changes and the quotes as well?

File: tools/notification-configuration/ci.yml [hunk 4]
```diff
@@ -25,3 +25,4 @@ extends:
   template: /eng/pipelines/templates/stages/archetype-sdk-tool-dotnet.yml
   parameters:
     ToolDirectory: tools/notification-configuration
+    SkipDotNetInstall: true
```
[the remark this answers is not in the document]
Comment (weshaggard) [reply tools/notification-configuration/ci.yml:28]: It makes things more consistent. If we just take what is on the machine we end up not working overtime.

File: tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs (hunk 3 above)
Comment (chidozieononiwu) [reply tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs:41]: Because I switched to netstandard. The syntax is a bit different, I guess for backward compatibility.

File: tools/code-owners-parser/ci.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+trigger:
+  branches:
+    include:
+      - main
+      - feature/*
+      - release/*
+      - hotfix/*
+  paths:
+    include:
+      - tools/notification-configuration
```
[the remark this answers is not in the document]
Comment (chidozieononiwu) [reply tools/code-owners-parser/ci.yml]: CreateRuleFabricBot has it own pipeline. I added this for the CodeOwnersParser

File: tools/notification-configuration/identity-resolver/Program.cs [hunk 6]
```diff
@@ -51,22 +54,27 @@ string targetvar
                         loggerFactory.CreateLogger<GitHubNameResolver>()
                     );
 
-                    var result = await githubNameResolver.GetMappingInformationFromAADName(identity);
+                    var result = await githubNameResolver.GetMappingInformationFromAADName(identityName);
+
+                    if (result == default(IdentityDetail))
```
Comment (weshaggard) [inline tools/notification-configuration/identity-resolver/Program.cs:59]:
```suggestion
                    if (result == default(IdentityDetail) && identityEmail != null)
```

File: tools/code-owners-parser/ci.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# NOTE: Please refer to https://aka.ms/azsdk/engsys/ci-yaml before editing this file.
```
Comment (weshaggard) [inline tools/code-owners-parser/ci.yml:1]: Do we need a ci for this library, as I assume we aren't going to publish it? It might be better to just build it as part of the other consuming library CI pipelines.

Comment (weshaggard) [reply tools/code-owners-parser/ci.yml:1]: OK. I wouldn't publish that until we figure out if we have any consumer for it though.

File: tools/notification-configuration/identity-resolver/Program.cs (hunk 6 above)
Comment (chidozieononiwu) [reply tools/notification-configuration/identity-resolver/Program.cs:59]: What value does the comparison to null add?

File: tools/notification-configuration/ci.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (chidozieononiwu) [reply tools/notification-configuration/ci.yml]: I specified the required version instead.

File: tools/notification-configuration/github-codeowner-subscriber/Program.cs [hunk 8]
```diff
@@ -129,17 +130,15 @@ class Program
 
                         var process = group.Pipeline.Process as YamlProcess;
 
-                        // Find matching contacts for the YAML file's path
-                        var parser = new CodeOwnersParser(codeownersContent);
                         logger.LogInformation("Searching CODEOWNERS for matching path for {0}", process.YamlFilename);
-                        var contacts = parser.GetContactsForPath(process.YamlFilename);
+                        var codeOwnerEntries = CodeOwnersFile.ParseContent(codeownersContent);
+                        var contacts = codeOwnerEntries.FindLast(x => x.PathExpression.Trim('/').StartsWith(process.YamlFilename.Trim('/'))).Owners;
```
Comment (weshaggard) [inline tools/notification-configuration/github-codeowner-subscriber/Program.cs:135]: This might be worth adding as a helper to the code owner parser at some point.
